library/Zend/Http/Client/Adapter/Proxy.php

```diff
@@ -297,10 +297,8 @@ protected function connectHandshake(
         // If all is good, switch socket to secure mode. We have to fall back
         // through the different modes
         $modes = array(
-            STREAM_CRYPTO_METHOD_TLS_CLIENT,
-            STREAM_CRYPTO_METHOD_SSLv3_CLIENT,
-            STREAM_CRYPTO_METHOD_SSLv23_CLIENT,
-            STREAM_CRYPTO_METHOD_SSLv2_CLIENT
+            // TODO: Add STREAM_CRYPTO_METHOD_TLSv1_3_CLIENT in the future when it is supported by PHP
```

Comment:
Must we really force TLS 1.2 on the client side?

Reply:
https://www.pcisecuritystandards.org/documents/Migrating_from_SSL_Early_TLS_Information%20Supplement_v1.pdf

> The best response is to disable SSL entirely and migrate to a more modern encryption protocol, which at the time of publication is a minimum of TLS v1.1, although entities are strongly encouraged to consider TLS v1.2

@kandy , @piotrekkaminski agreed to TLS 1.2 for Magento 2.3.

As far as client vs server.  I skimmed through PCI DSS and that PCI information supplement I've linked to above, and I haven't seen it make a distinction for different versions based on whether or not we are creating the TLS or receiving it.  From what I can tell, it seems that PCI wants ALL TLS connects to be either TLS 1.1 or 1.2, but "strongly encourages" 1.2.  

I'm not a lawyer, and I've only skimmed through the document, so if I am misinterpreting this, please let me know and we can make corrections.